### PR TITLE
Clean up and optimize sbrk()

### PR DESCRIPTION
**Pull request by kripken**
After we no longer include malloc/free by default (#12213) we can remove the
hack in sbrk which avoided using static memory. This gives some nice code
size improvements.

However this can't be as simple as we'd hope because of dynamic linking
plus `SAFE_HEAP`. In that mode we may try to use `sbrk_val` before the
dynamic linker has set it up. This is because `SAFE_HEAP` is not aware of
dynamic linking - it just instruments all loads and stores, period, including
ones that happen during dynamic linking startup. To avoid ordering issues
there, initialize `sbrk_val` manually. I'm not totally happy with that, but not
sure I see a better solution?

### CODE REVIEW

File: system/lib/sbrk.c [hunk 1]
```diff
@@ -29,25 +29,17 @@
...
 
 intptr_t* emscripten_get_sbrk_ptr() {
-  extern size_t __heap_base;
-  intptr_t* sbrk_ptr = &sbrk_val;
-#if __EMSCRIPTEN_PTHREADS__
-  if (__c11_atomic_load((_Atomic(intptr_t)*)sbrk_ptr, __ATOMIC_SEQ_CST) == 0) {
-    __c11_atomic_store((_Atomic(intptr_t)*)sbrk_ptr, (intptr_t)&__heap_base, __ATOMIC_SEQ_CST);
-  }
-#else
+#ifdef __USING_EMSCRIPTEN_RELOCATABLE__
```
Comment (sbc100) [inline system/lib/sbrk.c:37]: You can just use `__PIC__` here instead I think.

**Review by sbc100 — COMMENTED** [review]
Awesome! Thanks for following up on this.

File: system/lib/sbrk.c (hunk 1 above)
Comment (kripken) [reply system/lib/sbrk.c]: Thanks! I was wondering if there was something, but couldn't find a `__has_feature()` option for it, but I guess I was looking in the wrong place...

**Comment by kripken** [comment]
Yes, looks like `__PIC__` works perfectly, nice.

@sbc100 did you have other comments?

**Review by sbc100 — APPROVED** [review]

**Pull request merged**